### ISSUE TITLE
Registration with bootstrap script fails

### ISSUE THREAD

**Issue by bontreger** [amayagil/InsightsCannedDemo64#1]
running canned_demo.yml playbook fails, all systems fail at running the bootstrap script with 
"You no longer have access to the repositories that provide these products.  It is important that you apply an active subscription in order to resume access to security and other critical updates. If you don't have other active subscriptions, you can renew the expired subscription.  ", "[\u001b[92mSUCCESS\u001b[0m], [2019-09-20 22:36:48], [/usr/bin/yum -y erase 'katello-ca-consumer-*' katello-agent gofer katello-host-tools katello-host-tools-fact-plugin], completed successfully.", "", "[\u001b[92mSUCCESS\u001b[0m], [2019-09-20 22:36:48], [Removing /etc/rhsm/ca/katello-server-ca.pem], completed successfully.", "[NOTIFICATION], [2019-09-20 22:36:48], [System appears to be registered via another entitlement server. Attempting unregister] ", "[NOTIFICATION], [2019-09-20 22:36:48], [Unregistering] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:48], [/usr/sbin/subscription-manager unregister] ", "[\u001b[93mWARNING\u001b[0m], [2019-09-20 22:36:48], NON-FATAL: [/usr/sbin/subscription-manager unregister] failed to execute properly.", "Invalid credentials.", "Unregistering from: subscription.rhn.redhat.com:443/subscription", "", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:48], [/usr/sbin/subscription-manager clean] ", "All local data removed", "", "[NOTIFICATION], [2019-09-20 22:36:49], [Writing custom cURL configuration to allow download via HTTPS without certificate verification] ", "[NOTIFICATION], [2019-09-20 22:36:49], [Retrieving Client CA Certificate RPMs] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:49], [rpm -Uvh https://sat.example.com/pub/katello-ca-consumer-latest.noarch.rpm] ", "Retrieving https://sat.example.com/pub/katello-ca-consumer-latest.noarch.rpm", "Preparing...                          ########################################", "Updating / installing...", "katello-ca-consumer-sat.example.com-1.########################################", "[\u001b[92mSUCCESS\u001b[0m], [2019-09-20 22:36:53], [rpm -Uvh https://sat.example.com/pub/katello-ca-consumer-latest.noarch.rpm], completed successfully.", "", "[NOTIFICATION], [2019-09-20 22:36:53], [Deleting cURL configuration] ", "[\u001b[92mSUCCESS\u001b[0m], [2019-09-20 22:36:53], [Removing /tmp/tmpu0Kh7M], completed successfully.", "[NOTIFICATION], [2019-09-20 22:36:53], [Writing FQDN katello-fact] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:53], [Disassociating host id 392 for host ic9.example.com] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:53], [Deleting host id 392 for host ic9.example.com] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:56], [Calling Foreman API to create a host entry associated with the group & org] ", "[\u001b[92mSUCCESS\u001b[0m], [2019-09-20 22:36:56], [Successfully created host ic9.example.com], completed successfully.", "[NOTIFICATION], [2019-09-20 22:36:57], [Calling subscription-manager] ", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:57], [/usr/sbin/subscription-manager config --server.server_timeout=900] ", "", "", "[\u001b[94mRUNNING\u001b[0m], [2019-09-20 22:36:57], [/usr/sbin/subscription-manager register --org 'Default_Organization' --name 'ic9.example.com' --activationkey 'rhel7'  --serverurl=https://sat.example.com:443/rhsm --baseurl=https://sat.example.com/pulp/repos --force] ", "[\u001b[91mERROR\u001b[0m], [2019-09-20 22:36:57], EXITING: [/usr/sbin/subscription-manager register --org 'Default_Organization' --name 'ic9.example.com' --activationkey 'rhel7'  --serverurl=https://sat.example.com:443/rhsm --baseurl=https://sat.example.com/pulp/repos --force] failed to execute properly.", "Unable to register system, not all services available"]}

**Comment by idhaoui** [comment]
Hi,

I also faced issue registering this ic1.example.com.

./bootstrap.py -l admin -s sat.example.com -o "EXAMPLE.COM" -L Default_Location -g "rhel7" -a rhel7

Foreman Bootstrap Script
This script is designed to register new systems or to migrate an existing system to a Foreman server with Katello
admin's password:
[NOTIFICATION], [2019-10-30 03:02:45], [This system is not registered to RHN. Attempting to register via subscription-manager] 
[NOTIFICATION], [2019-10-30 03:02:45], [Checking subscription manager prerequisites] 
Loaded plugins: product-id, subscription-manager
[RUNNING], [2019-10-30 03:02:45], [/usr/bin/yum -y remove subscription-manager-gnome] 
Loaded plugins: product-id, search-disabled-repos, subscription-manager
No Match for argument: subscription-manager-gnome
No Packages marked for removal
[SUCCESS], [2019-10-30 03:02:45], [/usr/bin/yum -y remove subscription-manager-gnome], completed successfully.

[NOTIFICATION], [2019-10-30 03:02:45], [subscription-manager is installed already. Attempting update] 
[RUNNING], [2019-10-30 03:02:45], [/usr/bin/yum -y update subscription-manager 'subscription-manager-migration-*'] 
[WARNING], [2019-10-30 03:02:46], NON-FATAL: [/usr/bin/yum -y update subscription-manager 'subscription-manager-migration-*'] failed to execute properly.
Loaded plugins: product-id, search-disabled-repos, subscription-manager
There are no enabled repos.
 Run "yum repolist all" to see the repos you have.
 To enable Red Hat Subscription Management repositories:
     subscription-manager repos --enable <repo>
 To enable custom repositories:
     yum-config-manager --enable <repo>

[RUNNING], [2019-10-30 03:02:46], [/usr/bin/yum -y update yum openssl python] 
[WARNING], [2019-10-30 03:02:46], NON-FATAL: [/usr/bin/yum -y update yum openssl python] failed to execute properly.
Loaded plugins: product-id, search-disabled-repos, subscription-manager
There are no enabled repos.
 Run "yum repolist all" to see the repos you have.
 To enable Red Hat Subscription Management repositories:
     subscription-manager repos --enable <repo>
 To enable custom repositories:
     yum-config-manager --enable <repo>

[NOTIFICATION], [2019-10-30 03:02:46], [System appears to be registered via another entitlement server. Attempting unregister] 
[NOTIFICATION], [2019-10-30 03:02:46], [Cleaning old yum metadata] 
[RUNNING], [2019-10-30 03:02:46], [/usr/bin/yum -y clean all] 
[ERROR], [2019-10-30 03:02:47], EXITING: [/usr/bin/yum -y clean all] failed to execute properly.
Loaded plugins: product-id, search-disabled-repos, subscription-manager
There are no enabled repos.
 Run "yum repolist all" to see the repos you have.
 To enable Red Hat Subscription Management repositories:
     subscription-manager repos --enable <repo>
 To enable custom repositories:
     yum-config-manager --enable <repo>

**Comment by amayagil** [comment]
Hi,

looks like the manifest has expired.

I am not maintaining that demo anymore, the MBU folks should.

ATB,
Amaya

Amaya Rosa Gil Pippino

EMEA SENIOR SOLUTIONS ARCHITECT RHCA Level V

Infrastructure, Automation & Cloud.

Red Hat
<https://www.redhat.com>

amaya@redhat.com    T: 34-91-414-8861    M: +34-667-166-140
<https://red.ht/sig>

<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>

**Comment by robertodijo** [comment]
Anyone, found a solution? I am running into the same issue.

**Comment by amayagil** [comment]
Hi,

looks like the manifest has expired.

I am not maintaining that demo anymore, the MBU folks should.

ATB,
Amaya

Amaya Rosa Gil Pippino

EMEA SENIOR SOLUTIONS ARCHITECT RHCA Level V

Infrastructure, Automation & Cloud.

Red Hat
<https://www.redhat.com>

amaya@redhat.com    T: 34-91-414-8861    M: +34-667-166-140
<https://red.ht/sig>

<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>

**Comment by musman-ozzy** [comment]
Refreshing the Manifest and then adding the new visible subs to the activation key resolved the issue for me.

**Comment by amayagil** [comment]
Yes, that's the solution, but I'm no longer the maintainer, but the folks
at the MBU.

ATB,
Amaya

Amaya Rosa Gil Pippino

EMEA SENIOR SOLUTIONS ARCHITECT RHCA Level V

Infrastructure, Automation & Cloud.

Red Hat
<https://www.redhat.com>

amaya@redhat.com    T: 34-91-414-8861    M: +34-667-166-140
<https://red.ht/sig>

<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>
<https://red.ht/sig>
